From 38656c0b03fb7fcd3ad6666dca8b0708edfef921 Mon Sep 17 00:00:00 2001
From: hari gowtham <hgowtham@redhat.com>
Date: Mon, 23 Nov 2015 11:58:52 +0530
Subject: [PATCH 134/138] Tiering: change of error message for "v tier <vname>  detach status"

If the volume was not a tiered volume then empty status was being
printed instead of an error message.

Change-Id: I13ccb16e1562966976a48d9365ced4c8a124de59
BUG: 1236020
Signed-off-by: hari gowtham <hgowtham@redhat.com>
Reviewed-on: http://review.gluster.org/12713
Tested-by: NetBSD Build System <jenkins@build.gluster.org>
Tested-by: Gluster Build System <jenkins@build.gluster.com>
Tested-by: Dan Lambright <dlambrig@redhat.com>
Reviewed-by: Dan Lambright <dlambrig@redhat.com>
Signed-off-by: Hari Gowtham <hgowtham@redhat.com>
Reviewed-on: https://code.engineering.redhat.com/gerrit/62435
---
 cli/src/cli-rpc-ops.c                          |    4 ++--
 xlators/mgmt/glusterd/src/glusterd-rebalance.c |    8 ++++++++
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/cli/src/cli-rpc-ops.c b/cli/src/cli-rpc-ops.c
index f895375..9dc18a6 100644
--- a/cli/src/cli-rpc-ops.c
+++ b/cli/src/cli-rpc-ops.c
@@ -2366,10 +2366,10 @@ gf_cli_detach_tier_status_cbk (struct rpc_req *req, struct iovec *iov,
         ret = rsp.op_ret;
         if (rsp.op_ret == -1) {
                 if (strcmp (rsp.op_errstr, ""))
-                        snprintf (msg, sizeof (msg), "volume detach-tier %s: "
+                        snprintf (msg, sizeof (msg), "volume tier detach %s: "
                                   "failed: %s", cmd_str, rsp.op_errstr);
                 else
-                        snprintf (msg, sizeof (msg), "volume detach-tier %s: "
+                        snprintf (msg, sizeof (msg), "volume tier detach %s: "
                                   "failed", cmd_str);
 
                 if (global_state->mode & GLUSTER_MODE_XML)
diff --git a/xlators/mgmt/glusterd/src/glusterd-rebalance.c b/xlators/mgmt/glusterd/src/glusterd-rebalance.c
index 431e216..82dd6ee 100644
--- a/xlators/mgmt/glusterd/src/glusterd-rebalance.c
+++ b/xlators/mgmt/glusterd/src/glusterd-rebalance.c
@@ -740,6 +740,14 @@ glusterd_op_stage_rebalance (dict_t *dict, char **op_errstr)
                                 goto out;
                         }
                 }
+                if  (strstr(cmd_str, "tier") != NULL) {
+                        if (volinfo->type != GF_CLUSTER_TYPE_TIER) {
+                                snprintf (msg, sizeof(msg), "volume %s is not "
+                                          "a tier volume.", volinfo->volname);
+                                ret = -1;
+                                goto out;
+                        }
+                }
                 if (strstr(cmd_str, "detach-tier") != NULL) {
                         if (volinfo->type != GF_CLUSTER_TYPE_TIER) {
                                 snprintf (msg, sizeof(msg), "volume %s is not "
-- 
1.7.1

